### PR TITLE
start to move to warframe-worldstate-parser for data parsing, leverage hubot-markdown for markdown configuration

### PR DESCRIPTION
**Pull request by TobiTenno**

### CODE REVIEW

**Comment by TobiTenno** [comment]
resolve #27, #28, #29

**Comment by TobiTenno** [comment]
started #32

File: src/ondemand.coffee [hunk 1]
```diff
@@ -231,119 +86,185 @@ module.exports = (robot) ->
...
-                                    conclaveInstructAll, dsUtil.lineEnd, 
-                                    conclaveInstructWeekly, dsUtil.lineEnd, 
-                                    conclaveInstructDaily, dsUtil.blockEnd)
+                                    md.codeMulti, md.lineEnd, 
+                                    conclaveInstructAll, md.lineEnd, 
+                                    conclaveInstructWeekly, md.lineEnd, 
+                                    conclaveInstructDaily, md.blockEnd)
+  robot.respond /damage/, (res) ->  
+    damageURL = 'http://morningstar-wf.com/chart/Damage_2.0_Resistance_Flowchart.png'
+    res.send "${md.codeMulti}#{md.linkBegin}Damage 2.0#{md.linkMid}#{damageURL}#{md.linkEnd}#{md.blockEnd}"
```
Comment (nspacestd) [inline src/ondemand.coffee]: This should be `#{md.codeMulti}` instead of `${md.codeMulti}`

**Comment by nspacestd** [comment]
Some of the default markdown settings (namely BLOCK_END and CODE_BLOCK) don't work very well with telegram (and possibly other platforms too). I think those two should be left empty by default (or removed from alerts/news/etc...). I've opened a pull request [here](https://github.com/aliasfalse/hubot-markdown/pull/3)

md1
md2

File: src/ondemand.coffee [hunk 2]
```diff
@@ -231,119 +86,185 @@ module.exports = (robot) ->
...
+    userDB.getPlatform res.message.room, (err, platform) ->
+      if err
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getUpdatesString()), 300
+
   robot.respond /rewards/, (res) ->
-    res.send util.format('%sMission rewards%shttp://rewards.morningstar-wf.com%s',
-                           dsUtil.linkBegin, dsUtil.linkMid, dsUtil.linkEnd)
+    res.send "#{md.codeMulti}#{md.linkBegin}Mission rewards #{md.linkMid} http://rewards.morningstar-wf.com#{md.linkEnd}#{md.blockEnd}"
```
Comment (nspacestd) [inline src/ondemand.coffee]: Additional spaces break markdown formatting, should be `...rewards#{md.linkMid}http://...`

File: src/ondemand.coffee [hunk 3]
```diff
@@ -231,119 +86,185 @@ module.exports = (robot) ->
...
+      if err
+        return robot.logger.error err
+      checkWorldstate platform
+      setTimeout (-> res.send worldStates[platform].getSortieString()), 300
+        
+  robot.respond /tutorial(.+)/, (res) ->
+    tutorial = res.match[1]
+    focusReg = /\sfocus/
+    if focusReg.test tutorial
+      res.send "#{md.codeMulti}#{md.linkBegin}Warframe Focus #{md.linkMid} https://www.youtube.com/watch?v=IMltFZ97oXc #{md.linkEnd}#{md.blockEnd}"
```
Comment (nspacestd) [inline src/ondemand.coffee]: Same as above, additional spaces break markdown formatting. Should be `...Focus#{md.linkMid}https://...#{md.linkEnd}`

File: src/ondemand.coffee (hunk 1 above)
Comment (TobiTenno) [inline src/ondemand.coffee]: Indeed. I'll fix it. I hadn't had a problem on it because it apparently still works,  somehow, but I'll fix it.

**Comment by TobiTenno** [comment]
Encompasses #33, #32, #29, #27

**Pull request merged**